transcom/mymove
# B-19780 INT CSV Export for Queue Tables

## [B-19780](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19780)
> [!NOTE]  
> ~Minor additional work will be needed to support CSV export on the queues visible to HQ users. This will be added once B-20011 B-20013 and B-20014 are PO accepted.~
## Summary

Is there anything you would like reviewers to give additional scrutiny?
- Between the duplication for the HQ role and the addition of the exportValue function to the columns arrays in each queue component I've been thinking it might make sense to move those to their own utility/shared file. Thoughts?

[This github post](https://github.com/react-csv/react-csv/issues/72#issuecomment-421160842) explains why the `<CSVLink/>` component is programmatically clicked. That was the best solution I could find with this library to fetch the data asynchronously and still set the file name.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app
2. Login as a Counselor (ideally in KKFA due to available test data)
3. Locate the "Export to CSV" button to the top right of your queue table, press it and make sure it exports a CSV that matches the data in the table. The file should contain every move in your queue, not just those on the current page.
4. Apply your choice of filters or sorting, export to CSV again and ensure the data exported matches with applied filter and/or sorting.
5. Repeat for the SC's PPM Closeout Queue.
6. Repeat for the TOO's Queue.
7. Repeat for the TIO's Queue.
8. Repeat for all queues visible to the HQ user.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
